include/swift/SIL/SILInstruction.h

```diff
@@ -693,6 +745,236 @@ inline SingleValueInstruction *SILNode::castToSingleValueInstruction() {
            inst->getKind() <= SILInstructionKind::Last_##ID;    \
   }
 
+/// A value base result of a multiple value instruction.
+///
+/// *NOTE* We want this to be a pure abstract class that does not add /any/ size
+/// to subclasses.
+class MultipleValueInstructionResult : public ValueBase {
+  // Bits 0-2 of the first byte of SubclassData is used for the
+  // ValueOwnershipKind.
+  static constexpr unsigned NumBitsForOwnershipKind = 3;
+  static constexpr uint8_t ValueOwnershipKindMask =
+      uint8_t((1 << NumBitsForOwnershipKind) - 1);
+  static_assert(unsigned(ValueOwnershipKind::LastValueOwnershipKind) <
+                    (1 << NumBitsForOwnershipKind),
+                "LastValueOwnershipKind is larger than 3 bits?!");
+
+  // Bit 3 of the first byte of SubclassData is used to debug initialization by
+  // marking the value as fully initialized.
+  //
+  // The key thing here is that getParent() can not be called safely before the
+  // MultipleValueInstructionResultTrailingObjects containing the result has
+  // been fully constructed.
+  static constexpr uint8_t IsFullyInitMask = uint8_t(1)
+                                             << NumBitsForOwnershipKind;
```

Comment:
I just really want to make sure that people do not try to access getParent() in a constructor since the parent pointer may not be in place yet. That is the point of this. I should elaborate in the comment.

Reply:
Can you explain why the parent pointer is being initialized in two stages in the first place?  It seems unnecessary.

Reply:
I am not doing it in two stages. I am first initializing the parent data and then the data pointers. I was just being paranoid. That being said, I don't think it makes sense to burn a bit for this reason.

Reply:
*upon further reflection.